### DISCUSSION TITLE
zh/guide/drivers/local

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#16] [Announcements]
# zh/guide/drivers/local

AList v3的文档

https://alist.nn.ci/zh/guide/drivers/local.html

**Comment by zzerosama** [comment]
群晖怎么挂载本地存储？没有输账号密码的地方，总是提示不被允许，想挂载本地利用其他移动端设备从NAS里复制文件到其他网盘里

**Reply by ngol233** [reply]
群晖右键要挂载的文件夹，属性-权限-添加-搜索alist3-权限全打钩-确认，就有权限读取了。

**Reply by zzerosama** [reply]
感谢大佬，能用本地了，但是发现跟用WebDAV效果似乎一样，传的时候数据先从群晖里出来传出来写入缓存到移动设备，然后再传到的网盘，我其实想要的是在不需要进群晖系统操作界面的情况下，直接手机操作把群晖的文件直传到网盘，只占用上传，不绕路占别的任何磁盘写入和网络下载🤣

**Reply by jadetexy** [reply]
权限勾选后，在alist本地存储的 根文件夹的路径 该填什么，测试/volume2/新建文件夹 不行

**Reply by zzerosama** [reply]
> 权限勾选后，在alist本地存储的 根文件夹的路径 该填什么，测试/volume2/新建文件夹 不行

路径没错啊，不过你这个新建文件夹是确实存在的共享文件夹吗？什么也不输也应该能进去

**Reply by ngol233** [reply]
右键文件夹，属性，位置。
就填这个路径就行。

**Reply by jadetexy** [reply]
如果是什么也不输 " /  "在 根路径的情况下，加载出来的是8G的移动U盘的文件，并没有加载nas里的硬盘文件，是因为蜗牛的原因么

**Reply by jadetexy** [reply]
alist 在nas里是用docker安装的

**Reply by zzerosama** [reply]
> 如果是什么也不输 " /  "在 根路径的情况下，加载出来的是8G的移动U盘的文件，并没有加载nas里的硬盘文件，是因为蜗牛的原因么

我是直接输“/”就是这样，所以东西连系统文件都显示，我是群晖用的alist套件，docker应该也一样吧![屏幕截图 2022-10-17 144245.png](https://user-images.githubusercontent.com/57804509/196107179-be35d615-5a98-4760-b3aa-926d016f134a.png)![屏幕截图 2022-10-17 144606.png](https://user-images.githubusercontent.com/57804509/196107318-5332d47a-5d4c-49a7-91c4-cfc9ed3a3af4.png)

**Reply by ngol233** [reply]
不是很建议docker部署alist。映射的话也可以，但是不建议。
直接在群晖套件里面添加套件源，我的是6.3。用的这个源:https://spk.imnks.com/
然后在群晖套件里面搜索就行，包括nastool啊，aria2啊，alist，ddns等等，这个源都都有，alist能不用docker尽量不要用。

**Reply by ngol233** [reply]
然后alist挂载本地存储，我之前写过了，群晖文件管理，右键要挂载的文件夹，属性，位置，填进去就行。

**Reply by zzerosama** [reply]
> 然后alist挂载本地存储，我之前写过了，群晖文件管理，右键要挂载的文件夹，属性，位置，填进去就行。

他是蜗牛的，不是群晖，群晖完美解决

**Reply by drw2005** [reply]
群晖文件管理，右键要挂载的文件夹，属性，位置，填进去就行。
请问，为啥我按照上面解释填进去，打开提示下面：
failed get objs: failed to list objs: open /volume1/video: permission denied
这个提示是指哪里的权限啊？

**Reply by zzerosama** [reply]
> 群晖文件管理，右键要挂载的文件夹，属性，位置，填进去就行。
> 请问，为啥我按照上面解释填进去，打开提示下面：
> failed get objs: failed to list objs: open /volume1/video: permission denied
> 这个提示是指哪里的权限啊？

属性-权限里，搜索alist，给它权限，意思就是alist这个应用也属于一个用户，它是一个内部应用用户，在文件夹权限里加入上，alist才能有访问这个本地文件夹的权限

**Reply by drw2005** [reply]
可以了，感谢解答！

**Reply by ocean521** [reply]
威联通怎么操作啊，谢谢

**Reply by zzerosama** [reply]
> 威联通怎么操作啊，谢谢

没用过威联通啊😂

**Comment by zzerosama** [comment]
有用群晖的大佬知道怎样用这个挂载的本地存储从群晖直传到网盘并且不读写磁盘的方法吗？
初衷是想用手机ES文件浏览器挂载Alist，操作挂载群晖本地存储的文件直传到阿里云，但每次传发现数据都要到手机里走一圈缓存再上传

**Comment by caodapengrom** [comment]
挂载了群晖的本地文件，pdf不能在线预览怎么解决啊？出现pdf.js错误PDF.js v3.0.44 (build: 6852001b0)
信息：Failed to fetch

**Reply by gjssss** [reply]
建议看一下常见问题，应该是需要https

**Comment by ocean521** [comment]
请问威联通QNAP如何挂载阿里云至本地存储？想使用威联通的PLEX看阿里云盘内的影视文件。现在只是通过HYBIRDMOUNT，将阿里云盘挂载到“与我共享”里面，PLEX浏览不到这个文件夹。如何挂载到/mnt之类的NAS本地文件夹啊，谢谢赐教、

**Reply by ocean521** [reply]
更正一下目前可以“通过HYBIRDMOUNT，将阿里云盘挂载到“webdav cloud/server”里面”。但是，PLEX浏览不到这个文件夹。
问题：
如何将云盘挂载到/mnt之类的NAS本地文件夹，以便PLEX浏览，谢谢赐教、

**Reply by ocean521** [reply]
使用rclone的web ui 及命令行，都可以实现将阿里云盘挂载到本地文件夹。问题是每次重启都需要重新挂载，哪位大神可以指导一下计划任务的操作。

**Reply by zyuzone** [reply]
我是用 https://github.com/iranee/qnap-alist-webdav/ 这个alist的qnap套件挂载阿里云盘，然后HybridMount挂载为本地文件夹。目前一切正常（包括file station），emby也能正常使用，速度还挺快，进度条拖动很流畅。

**Reply by Ethanwish** [reply]
> 我是用 https://github.com/iranee/qnap-alist-webdav/ 这个alist的qnap套件挂载阿里云盘，然后HybridMount挂载为本地文件夹。目前一切正常（包括file station），emby也能正常使用，速度还挺快，进度条拖动很流畅。

我试了一下，没找到正确的方式，你是使用的云网关吗还是普通CIFS挂载到本地的？谢谢传授下教程

**Reply by maotui** [reply]
威联通的办法，仅供参考，我也是刚刚用上，不知道是否稳定。
1、用HybridMount挂载“云硬盘装载”后，可以从File station文件管理器中看到“webdav cloud/server”。
2、使用emby的新建媒体库功能，查找这个目录：/share/external/.cm/，在这个目录中继续进入子目录，大概2级子目录，你就会找到alist挂载好的网盘文件夹了，随后复制该路径，黏贴到plex，就可以扫库刮削了。
3、备注：我之前遇到过一次问题，plex中影片刮削的内容、封面都在，但无法关联网盘中的视频原文件，后来重新清扫，整理。不知道能用多长时间。

**Comment by sonzi999** [comment]
openwrt怎么挂载本地存储？/root出来的不是本地真实的

**Comment by sing0626** [comment]
为啥rclone没了，用不了了，帮忙搞搞一下，急！

**Comment by jadetexy** [comment]
管理员登录的alist    无法对挂载点nas文件进行增删操作 提示   permission denied  是怎么回事呀

**Comment by simplove** [comment]
https://file.xxx.me/d/loc/file/photo-pichuli.rar
本地存储的文件直链没办法被CDN缓存啊，试过了cf和百度云加速都不可以缓存。文件实际只有不到2M

**Comment by makergit365** [comment]
docker的挂本地存储有没有详细介绍，我给映射的目录弄不上去，Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path

**Comment by ponponon** [comment]
怎么添加存储？一打开就是 `failed get storage: can't find storage with rawPath: /`

**Reply by 1203952894** [reply]
首先你需要用 `docker exec -it alist ./alist admin
` 获取管理员信息然后进行相关的配置在  https://www.youtube.com/watch?v=uoVIBtpkOms          YouTube 上有详细的视频演示希望对你有用

**Comment by shmilyabyy** [comment]
老哥想问下，docker安装的alist如何设置本机存储。直接/root是容器里的/root吗？

**Reply by makergit365** [reply]
/就直接是alist的根目录，下级目录是//nas，你直接/看一下就懂了

**Reply by shmilyabyy** [reply]
明白了

**Reply by sinaamos** [reply]
弄成功了吗，我也是docker

**Reply by xianghalo** [reply]
docker的存储空间添加映射，如（存储空间2/downloads 装载路径就 /root ）,设置好后在alist的存储设置中，本机存储设置里，根文件夹路径：/root  ，就可以了。

**Comment by hlf2016** [comment]
大佬 这个本地存储为啥不支持密码查看呢 方便支持下吗 毕竟在自己服务器上的东西 是要走带宽流量的

**Comment by jadetexy** [comment]
挂载alist访问是  群晖IP+5244，如果群晖的IP本身是带端口的例如3000，怎么办，IP:3000:5244？

**Comment by Deleted-Account** [comment]
请问如何设置密码访问？我挂载了软路由OpenWRT下的一个目录，任何人打开Alist的网址就能直接看到，无需账号密码

---

自问自答，把guest的目录换一个就好

**Comment by a630111** [comment]
要如何才能把群暉上的資料夾掛在ALIST上面?

**Comment by sindwerra** [comment]
用docker进行安装之后本地挂载应该如何配置呢？root folder选项尝试添加了本地电脑路径但是提示“Failed init storage but storage is already created: failed init storage: root folder xxx path does not exists”，然后发现root folder path填alist的docker镜像里的路径是可以创建成功的。。。

**Reply by ykck2** [reply]
容器是虚拟的，你要让容器有权限访问宿主机，同时卷也是要映射到容器内的。宿主机包含容器，这个（root folder path填alist的docker镜像里的路径是可以创建成功）肯定是可以的

**Comment by 771319205** [comment]
请问我想访问macbook桌面上的一个文件夹，请问怎么填路径，用终端提取出来的路径也不能用

**Comment by dutLQG** [comment]
mac里面挂外接的硬盘挂不上本地存储，挂根目录里面没有volumes选项（软链接映射），如何解决

**Comment by hnkfly2006** [comment]
看评论区一片哀嚎，同样是卡在挂载群晖本机存储上面，用“/”出现的应该是docker的根目录， 用“/root”出现的是空白，怎么才能挂载群晖存储空间里的目录呢，有没有设置成功的大佬说一下

**Reply by ykck2** [reply]
容器时虚拟的，你要让容器有权限访问宿主机，同时卷也是要映射到容器内的。

**Reply by wanglei52076** [reply]
能详细说一下吗 没太看懂！

**Comment by ykck2** [comment]
docker run -d --restart=always -v /etc/alist:/opt/alist/data -p 5244:5244 -e PUID=0 -e PGID=0 -e UMASK=022 --name=&quot;alist&quot; xhofe/alist:latest 这个拉取镜像中的 -v 后面就你主机地址与容器内地址的映射，以冒号分隔，你要容器读自己的硬盘，就要把你硬盘的挂载目录映射进容器。 



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年3月22日(星期三) 下午4:24
收件人: "alist-org/docs";
抄送: "Comment";
主题: Re: [alist-org/docs] zh/guide/drivers/local (Discussion #16)





 
能详细说一下吗 没太看懂！
 
&mdash;

**Comment by hxinbeng** [comment]
直接下就报错---[{"code":401,"message":"expire missing","data":null}]
使用下面的批量下载就能正常下载

使用挂载本地存储和WebDav都会出现,其它用网盘的就不会

**Comment by chen20220801** [comment]
failed get objs: failed to list objs: Post "/api/public/path": unsupported protocol scheme ""

**Comment by zxf09050** [comment]
alist安装在手机上：添加本机存储 根文件路径填/sdcard，成功。局域网其他设备访问不了，500

**Reply by zxf09050** [reply]
好了。给手機Termux存储权限就好了。

**Comment by skymailwu** [comment]
请问哪个使用签名是什么意思？有什么作用？

**Comment by shenliyu** [comment]
大佬求救：

我是用的Windows Server 2022，初衷和规划是：绝大多数的电影放在阿里云，毕竟外网访问快一点，目前用“阿里云Open”访问没有问题，还有一些电影是在阿里云没法存储的，放在本地，然后用“本地存储”，再用“别名”将两个聚合起来，然后挂infuse。

现在遇到的问题是阿里云内的文件访问一切正常，反而是本地的文件，可以查看列表，但无法打开文件；目前测试，除了wav音频，txt文本，其它的文件都无法访问内容，视频无法播放，无论在浏览器还是infuse，图片无法查看，pdf无法预览....

请问有大佬知道咋回事不，感谢。

**Comment by lztttt** [comment]
![Snipaste_2023-07-30_14-27-32.png](https://img1.imgtp.com/2023/07/30/NE63tWv3.png)
Debian下把U盘挂载在一个目录下

**Reply by lztttt** [reply]
但是尝试在alist中挂载，显示目录不存在

**Comment by cmhr0086** [comment]
在下载的时候，头是127.0.0.1:5244，怎么把这个改成外网域名或者IP？

**Comment by yuunie** [comment]
# 已经启动容器后，在 docker 中挂载宿主机目录

1、查看 docker 容器 ID，并记录下 ID
```
docker ps
```

2、停止 docker 服务
```
sudo systemctl stop docker.socket
sudo systemctl stop docker.service
```

3、进入 docker 容器配置目录
```
cd /var/lib/docker/containers
cd 【第一步获得的容器 ID】
```

4、编辑容器配置文件
  * 这里将用到 `vim` 没有的先安装
```
vim config.v2.json
```
  * 复制所有内容出来，然后格式化 JSON
  * 将 JSON MountPoints 中的对象复制一份，修改 KEY 名，修改 Source 对应的路径为宿主机目录，修改 Destination 路径为容器内目录，修改 Spec 中 Source 和 Source 一致， Spec 中 Target 和 Destination 一致
  * 压缩 JSON （可以找个在线的 JSON 压缩工具）
  * 回到 config.v2.json 文件，按 `dd` 删除内容
  * 粘贴刚复制的压缩后的 JSON
  * `Esc`
  * `:wq` 回车

5、编辑容器配置文件
```
vim hostconfig.json
```
  * 步骤和 4 一致，这里修改的东西不同
  * 修改 Binds，Binds 默认应该是 `null`
  * 修改 `null` 为 `[ "这里为步骤 4 中 Source":"这里为步骤 4 中 Destination" ]`
  * 例子：`"Binds":["/home/share:/home"]`

6、重新启动容器
```
sudo systemctl daemon-reload
sudo systemctl restart docker.service
```

7、查看容器是否正常
```
docker ps
```
  * 查看容器是否正常启动
```
docker exec -it 【这里写容器 ID 或名称 】 bash
```
  * 进入容器后使用 `ls` 查看刚挂载的目录是否成功

8、挂载成功之后，就可以通过本地存储将目录添加到 alist 中了

**Reply by ag12530** [reply]
你能不能发一个改好的config.v2.json里面内容上来看看？？？
或者发改前和改后的对比上来看看！

**Comment by chuyu5762** [comment]
使用本地储存，pdf无法预览

**Comment by wolfliu13588** [comment]
在启动docker的时候，PUID和PGID都是0，挂载本地存储后，可以进入/home和/mnt这两个目录，但/home和/mnt这两个目录下面的目录和文件看不到，没有提示：permission denied，不知道是什么原因。

**Reply by wolfliu13588** [reply]
我发现这个本机存储是挂载的容器里面的文件夹，难怪看不到原生系统里面/home和/mnt里面的文件，不知道应该怎么挂载原生系统的文件夹呢？

**Reply by ag12530** [reply]
我是使用casaos挂在的，直接界面操作，很方便！

**Comment by joshua644** [comment]
请问Windows下怎么挂载D盘或者其他盘的文件夹？路径应该怎么写？

**Reply by xuanxuan1231** [reply]
`D:`

**Comment by wolfliu13588** [comment]
我用的是linux，然后alist运行在docker里面，所以挂载的情况和windows版的不一样，无法帮到你，抱歉！发自我的 iPhone在 2024年1月25日，15:09，joshua644 ***@***.***> 写道：﻿
请问Windows下怎么挂载D盘或者其他盘的文件夹？路径应该怎么写？

**Comment by duzengjie** [comment]
我使用的是本地存储但是没有出现上传和新建的按钮

**Comment by wolfliu13588** [comment]
我个人觉得可能是权限的问题，因为你是windows安装的alist，我无法给出更多有效建议，抱歉！发自我的 iPhone在 2024年2月7日，09:21，我知道了嗯 ***@***.***> 写道：﻿
我使用的是本地存储但是没有出现上传和新建的按钮

**Comment by jiangjin19856** [comment]
分享一个视频缩略图的预览方法。

**Reply by jiangjin19856** [reply]
不能上传图片，只能用文字的方法写一下了。
1、用管理账号登录后台。
2、设置-预览-外部预览-加入代码。
{
   "mp4,mkv,avi,mov,rmvb,webm,flv":{
        "ffmpeg":"/opt/bin/ffmpeg"
  }
}
/opt/bin/ffmeg 为自己的ffmeg安装路径，请根据自己的实际情况更换。
3、其实代码的我也懂，就是抄的旁边的Iframe 预览的作业，个人理解前面的 mp4之类的是需要预览的扩展名，后面的是地址，有不对的地方请在大佬补充。

**Reply by lm317379829** [reply]
没用呀，本地挂载还是不显示视频缩略图。
ffmpeg路径没错

**Comment by 1254565651** [comment]
群晖怎么挂载本地存储？ 我挂载成功了 但是有很多隐藏的目录的 跟我在网页里看群晖的文件夹的目录不一样

**Comment by Dregxlx** [comment]
有谁知道alist怎么挂载mac本地路径

**Comment by lm317379829** [comment]
请问openwrt 安装 二进制alist文件怎么启用本地存储视频封面？具体信息如下，但是无法显示视频缩略图。
alist 为 v3.34.0 版
ffmpeg已安装至 /usr/bin/ffmpeg
本机存储已开启缩略图

**Reply by Muione** [reply]
使用以下命令查看`ffmpeg`是否添加到了PATH中：
``` bash
which ffmpeg
```
如果在的话则会输出`ffmpeg`的路径，比如：` /usr/bin/ffmpeg`.
在此基础上，查看本地存储中的设置中有没有设置`缩略图缓存文件夹`

**Reply by lm317379829** [reply]
都没问题，但是就是没有缩略图。
算了，我之后写个爬虫去刮削吧。
感谢解答

**Comment by 2055707266** [comment]
为什么只能挂载C盘，不能挂载D盘，我的另外一台电脑就正常，这个一直报错Failed init storage: root folder D:\ not exists

**Comment by jalousiex** [comment]
请教个权限问题，本地目录挂出来，是通过guest用户来控制webdav挂载后的权限吗

**Comment by p981277902** [comment]
请问一下，回收站的文件，没有有效期天数这样的设置吗？

**Comment by ljesy** [comment]
为啥 我挂本地的/volume1/tool 提示这个路径不存在 ，明明有这个文件夹的啊
老版的挂着，没问题，今天 装的最新版，提示:
Failed init storage but storage is already created: failed init storage: root folder /volume1/tool not exists

**Reply by liruicrecky** [reply]
我不知道你是怎么部署的，上面有docker的办法。
在compose里面，或者直接运行里面，先得挂载本地目录到alist的镜像里，然后直接挂载就好了，例如：
- '/volume1/download:/download'  （本机需要挂载的文件夹:alist镜像里的文件夹）

然后直接根文件夹路径挂在alist里的文件夹/download就可以了

**Comment by xiaoyu-success** [comment]
想问一下，我在A服务器上搭了alist用本地存储，在B云服务器上进行了反代，那么下载的时候会消耗B的流量吗？

**Comment by agilersu** [comment]
windows下在本地电脑挂载网络盘上文件夹，可以正常挂载但是无法访问，比如 \\SK-20200716DLVL\english；若是映射成本地盘符，比如X:\则无法挂载，请问如何操作呢？
